components/camel-hdfs/src/main/java/org/apache/camel/component/hdfs/DefaultHdfsFile.java

```diff
@@ -63,11 +64,8 @@ protected final long copyBytes(InputStream in, OutputStream out, int buffSize, b
     }
 
     protected final Writable getWritable(Object obj, Exchange exchange, Holder<Integer> size) {
-        Class<?> objCls = obj == null ? null : obj.getClass();
-        HdfsWritableFactories.HdfsWritableFactory objWritableFactory = WritableCache.writables.get(objCls);
-        if (objWritableFactory == null) {
-            objWritableFactory = new HdfsWritableFactories.HdfsObjectWritableFactory();
-        }
+        Class<?> objCls = Optional.ofNullable(obj).orElse(new UnknownType()).getClass();
```

Comment:
As far as I understood, you want to use some sort of empty class instead of null, do you think is better to use [`Void`](https://docs.oracle.com/javase/8/docs/api/java/lang/Void.html) class instead of defining your own?

Reply:
I created a new class because:

1. I thought that using Void would re-purpose another class.
2. UnknownType is the "_correct_"/label for that type.
3. Creating a new class is almost costless.

Would change it if there precedent of using Void for unknown map item in other components, to preserve consistency.

Reply:
Sure, I think it should be fine